### ISSUE TITLE
SPF.js doesn't return responses when we have multiple lines!

### ISSUE THREAD

**Issue by kamrava** [youtube/spfjs#438]
SPF.js doesn't work with this:

```
<link rel=\"stylesheet\" href=\"css/style1.css\">
<link rel=\"stylesheet\" href=\"css/style2.css\">
<link rel=\"stylesheet\" href=\"css/style3.css\">
```
But if I put all lines into one line it works fine. Like this:
`<link rel=\"stylesheet\" href=\"css/style1.css\"><link rel=\"stylesheet\" href=\"css/style2.css\"><link rel=\"stylesheet\" href=\"css/style3.css\">`

Why SPF.js doesn't work when in multiple lines?

**Comment by rviscomi** [comment]
White space doesn't affect the DOM, so I doubt that's what's causing your issue. Can you post a jsfiddle with an example of your code not working?

**Comment by kamrava** [comment]
I'm trying to integrating SPF.js with Laravel. Here is my structure:

> views/users-list/head.blade.php
> views/users-list/body.blade.php
> views/users-list/foot.blade.php
> views/users-list/index.blade.php

**head.blade.php**
```
<link rel="stylesheet" href="css/style1.css">
<link rel="stylesheet" href="css/style2.css">
<link rel="stylesheet" href="css/style3.css">
```
**body.blade.php**
```
<div class="row">
  <div class="col-md-12">
    <table id="UsersTable" class="table">
      <thead>
      <tr>
          <th>ID</th>
          <th>Last Name</th>
          <th>Email</th>
          <th>Mobile</th>
      </tr>
      </thead>
      <tbody>
        @foreach($users as $user)
        <tr>
          <td>{{ $user->id }}</td>
          <td>{{ $user->first_name }}</td>
          <td>{{ $user->email }}</td>
          <td>{{ $user->mobile }}</td>
        </tr>
        @endforeach
      </tbody>
  </table>
  </div>
</div>
```
**foot.blade.php**
```
<script src="/js/bootbox.min.js"></script>
<script src="/js/datatables.min.js"></script>
<script src="/js/datatables.bootstrap.js"></script>
```

**index.blade.php**
```
{
  "head": "@include('users-list.head')",
  "body": {
    "spf-posts-container": "@include('users-list.body')"
  },
  "foot": "@include('users-list.foot')"
}
```
As you can see everything is solid and clear in my structure but this is not working until I manually replace all ```"``` with ```\"``` and also put all lines into one line in each blade file(```head.blade.php```, ```body.blade.php```, ```foot.blade.php```)

Like so:

**head.blade.php**

`<link rel=\"stylesheet\" href=\"css/style1.css\"><link rel=\"stylesheet\" href=\"css/style2.css\"><link rel=\"stylesheet\" href=\"css/style3.css\">`

**body.blade.php**

`<div class=\"row\"><div class=\"col-md-12\"><table id=\"UsersTable\" class=\"table\"><thead><tr><th>ID</th><th>Last Name</th><th>Email</th><th>Mobile</th></tr></thead><tbody>@foreach($users as $user)<tr><td>{{ $user->id }}</td><td>{{ $user->first_name }}</td><td>{{ $user->email }}</td><td>{{ $user->mobile }}</td></tr>@endforeach</tbody></table></div></div>`

**foot.blade.php**

`<script src=\"/js/bootbox.min.js\"></script><script src=\"/js/datatables.min.js\"></script><script src=\"/js/datatables.bootstrap.js\"></script>`

But I don't like this! it makes my code messy ...

Any ideas?

**Comment by PhilHarnish** [comment]
I think the issue is in "index.blade.php". These lines:

```
{
  "head": "@include('users-list.head')",
  "body": {
    "spf-posts-container": "@include('users-list.body')"
  },
  "foot": "@include('users-list.foot')"
}
```

are instructing PHP to inject the contents of the other files, in line. So if "users-list/head.php" contained this string: `end of head", "new_attribute": "new_value` you would end up with this output:

```
{
  "head": "end of head", "new_attribute": "new_value",
  ...
```

Which of course is not what you intended. See if you can find a method in PHP for escaping string values for use in JSON. Perhaps this? http://php.net/manual/en/function.addslashes.php

**Comment by kamrava** [comment]
You are right about slashes. But I  can't understand why it only works when I put all lines into one line! If I don't it won't work! 
Any ideas?

**Comment by PhilHarnish** [comment]
Yes, same issue.

If "users-list/head.php" contained this:

```
line 1
line 2
line 3
```

You would end up with this output:

```
{
  "head": "line break 1: 
line break 2: 
line 3",
  "body": {
  ...
```

Which is _also_ invalid JSON (see "string" section of [JSON definition](http://www.json.org/)). If you're able to find a method to load a file in PHP and escape use the "addslashes" method it will also convert newline characters to something acceptable for a JSON string (which is the 2-character string "\n").

**Comment by kamrava** [comment]
Thank You, I've provided a Laravel Package. [laravel-spfjs](https://github.com/kamrava/laravel-spfjs)
:)
